src/en/developer-layers-interfaces.md

```diff
@@ -204,63 +193,68 @@ def configure_website(website):
 We're now ready to implement the requirer interface in `requires.py`
 
 ```python
-from charms.reactive import hook
-from charms.reactive import RelationBase
-from charms.reactive import scopes
-
-
-class HttpRequires(RelationBase):
-    scope = scopes.UNIT
-
-    @hook('{requires:http}-relation-{joined,changed}')
-    def changed(self):
-        conv = self.conversation()
-        if conv.get_remote('port'):
-            # this unit's conversation has a port, so
-            # it is part of the set of available units
-            conv.set_state('{relation_name}.available')
+from charms.reactive import when_any, when_not
+from charms.reactive import set_flag, clear_flag
+from charms.reactive import Endpoint
+
+
+class HttpRequires(Endpoint):
+    @when_any('endpoint.{relation_name}.changed.hostname',
+              'endpoint.{relation_name}.changed.port')
+    def new_website(self):
```

Comment:
To be clear, the `self` here is not specific to the hook/relation that triggered this handler, right? There is only one `HttpRequires` object for every endpoint using this interface, correct? We might want to make this explicit in the docs.